### ISSUE TITLE
Regex attribute validation doesn't work

### ISSUE THREAD

**Issue by emahuni** [strapi/strapi#14184]
### Required System information

- Node.js version: 14.20
- NPM version:
- Strapi version: 4.3.4
- Database: Postgres 12
- Operating system: MacOS 12.4

### Describe the bug

The regex validation doesn't work as intended. When you put a regex validation on an attribute it doesn't validate it properly when you enter data in the content manager.

### Steps to reproduce the behavior

1. create a content type and create a text field
2. click advanced settings
3. put regexp pattern /test/ as a validation for the text
4. save the content type
5. go to content builder and open the content type you just created
6. enter text "test"
7. you will get an error "The value does not match the regex." when you try to save

### Expected behavior

It should respect the the regexp properly

### Screenshots

![image](https://user-images.githubusercontent.com/7290508/186328674-448b7b06-c769-4211-8e6c-64f9d2acc731.png)

![image](https://user-images.githubusercontent.com/7290508/186328636-c3e6f618-f279-44e2-9270-bff3003a1c43.png)

**Comment by WalkingPizza** [comment]
You don't need the forward slashes when entering a pattern in the input field.
Just having `test` in there should fix your issue!

**Comment by gu-stav** [comment]
@WalkingPizza That sounds like a good solution to me.

**Comment by WalkingPizza** [comment]
@gu-stav a bit of an edge case, but do you think it'd be a breaking change for people who actually want to make sure strings begin and end with a forward slash?

**Comment by gu-stav** [comment]
@WalkingPizza I think we should normalize the string, so that for the user it doesn't matter whether `/` is included or not. We could also add a validation, checking if the string starts/ end with `/`: if so print an error. I'd need to check what our approach with the other fields is (permissive vs. strict).

**Comment by emahuni** [comment]
> You don't need the forward slashes when entering a pattern in the input field. Just having `test` in there should fix your issue!
> 
> Strapi Team: should we enhance this by allowing forward slashes in the input and then removing them in the entity validator?

Oh I see. Thanks, but I think that should be documented in the schema attributes validation. I tried looking it up.

**Comment by gu-stav** [comment]
@emahuni I agree. Where did you look for documentation if I may ask? I thinks it is worth adding there (along a better validation) ...

**Comment by jaysoni-ash42** [comment]
@gu-stav can i work on this issue.

**Comment by emahuni** [comment]
> @emahuni I agree. Where did you look for documentation if I may ask? I thinks it is worth adding there (along a better validation) ...

On the documentation site: https://docs.strapi.io/developer-docs/latest/development/backend-customization/models.html#validations

**Comment by Dule-mart** [comment]
> > @emahuni I agree. Where did you look for documentation if I may ask? I think it is worth adding there (along with a better validation) ...
> 
> On the documentation site: https://docs.strapi.io/developer-docs/latest/development/backend-customization/models.html#validations I thought it should be there. Then I searched for 'regexp' and nothing of relevance showed up, just some routes stuff.

Does this mean that there is a gap in the documentation? I can help with fixing the gap.

**Comment by gu-stav** [comment]
Hi. We have finally reviewed this issue today. I will suggest to the team to improve the validation of the regex to be permissive when it comes to `/`s at the start and end. I think we will also create a follow-up issue to improve the documentation.

**Comment by Dule-martins** [comment]
Open to join force in improving the documentation
